src/detect-ikev1-key-exchange-payload.c

```diff
@@ -0,0 +1,172 @@
+#define BUFFER_NAME_SERVER_KEY_EXCHANGE  "ikev1.server_key_exchange_payload"
+#define BUFFER_DESC_SERVER_KEY_EXCHANGE  "ikev1 server key_exchange payload"
+
+static int g_buffer_client_key_exchange_id = 0;
+static int g_buffer_server_key_exchange_id = 0;
+
+static int DetectClientKeyExchangeSetup(DetectEngineCtx *de_ctx, Signature *s, const char *str)
+{
+    if (DetectBufferSetActiveList(s, g_buffer_client_key_exchange_id) < 0)
+        return -1;
+
+    if (DetectSignatureSetAppProto(s, ALPROTO_IKEV1) < 0)
+        return -1;
+
+    return 0;
+}
+
+static int DetectServerKeyExchangeSetup(DetectEngineCtx *de_ctx, Signature *s, const char *str)
+{
+    if (DetectBufferSetActiveList(s, g_buffer_server_key_exchange_id) < 0)
+        return -1;
+
+    if (DetectSignatureSetAppProto(s, ALPROTO_IKEV1) < 0)
+        return -1;
+
+    return 0;
+}
+
+static InspectionBuffer *GetClientKeyExchangeData(DetectEngineThreadCtx *det_ctx,
+        const DetectEngineTransforms *transforms, Flow *_f,
+        const uint8_t _flow_flags, void *txv, const int list_id)
+{
+    InspectionBuffer *buffer = InspectionBufferGet(det_ctx, list_id);
+    if (buffer->inspect == NULL) {
+        const uint8_t *b = NULL;
+        uint32_t b_len = 0;
+
+        if (rs_ikev1_state_get_key_exchange(txv, &b, &b_len) != 1)
+            return NULL;
+        if (b == NULL || b_len == 0)
+            return NULL;
+
+        InspectionBufferSetup(buffer, b, b_len);
+        InspectionBufferApplyTransforms(buffer, transforms);
+    }
+
+    return buffer;
+}
+
+static InspectionBuffer *GetServerKeyExchangeData(DetectEngineThreadCtx *det_ctx,
+                                          const DetectEngineTransforms *transforms, Flow *_f,
+                                          const uint8_t _flow_flags, void *txv, const int list_id)
+{
+    InspectionBuffer *buffer = InspectionBufferGet(det_ctx, list_id);
+    if (buffer->inspect == NULL) {
+        const uint8_t *b = NULL;
+        uint32_t b_len = 0;
+
+        if (rs_ikev1_state_get_key_exchange(txv, &b, &b_len) != 1)
+            return NULL;
+        if (b == NULL || b_len == 0)
+            return NULL;
+
+        InspectionBufferSetup(buffer, b, b_len);
+        InspectionBufferApplyTransforms(buffer, transforms);
+    }
+
+    return buffer;
+}
+
+void DetectIkev1KeyExchangeRegister(void)
+{
+    // register client_key_exchange
+    sigmatch_table[DETECT_AL_IKEV1_CLIENT_KEY_EXCHANGE].name = KEYWORD_NAME_CLIENT_KEY_EXCHANGE;
+    sigmatch_table[DETECT_AL_IKEV1_CLIENT_KEY_EXCHANGE].url = "/rules/" KEYWORD_DOC_CLIENT_KEY_EXCHANGE
+    sigmatch_table[DETECT_AL_IKEV1_CLIENT_KEY_EXCHANGE].desc = "sticky buffer to match on the IKEv1 client_key_exchange_payload";
+    sigmatch_table[DETECT_AL_IKEV1_CLIENT_KEY_EXCHANGE].Setup = DetectClientKeyExchangeSetup;
+    sigmatch_table[DETECT_AL_IKEV1_CLIENT_KEY_EXCHANGE].flags |= SIGMATCH_NOOPT|SIGMATCH_INFO_STICKY_BUFFER;
```

Comment:
Can we validate the content is hexadecimal ?
like https://github.com/OISF/suricata/blob/a98a20973230bb662ddf204fb4be7b7fb657024c/src/detect-ssh-hassh.c#L213